crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -158,9 +158,21 @@ impl EventTimelineItem {
         let event_id = event.event_id().to_owned();
         let is_own = client.user_id().map(|uid| uid == sender).unwrap_or(false);
 
+        // Get the room's power levels for calculating the latest event
```

Comment:
Same here I guess but I'm not sure how heavy it actually is. Does fetching a room and power levels have any potential of introducing slowdowns? 

Passing the client into PossibleLatestEvent would be weird and it wouldn't improve the sliding sync part 🤔

Reply:
I *think* getting both the room and the state event should be quite lightweight, but I'm not sure how much it could affect performance. The alternative could be to only load them if we reach a knock state event, maybe, although I'd have to change some data structures.